src/components/TeacherComponents/hooks/useOnboardingStatus.ts

```diff
@@ -0,0 +1,40 @@
+import { useUser } from "@clerk/nextjs";
+import { useEffect, useRef, useState } from "react";
+
+/**
+ * Returns the onboarding status for the current session
+ *
+ * `loading` while Clerk is loading
+ * `not-onboarded` when there is no active user or the active user has not completed onboarding
+ * `onboarded` when the active user has completed onboarding
+ * `unknown` if Clerk fails to load in 10 seconds
+ */
+export function useOnboardingStatus() {
+  const { isLoaded, user } = useUser();
+  const [hasTimedout, setHasTimedout] = useState(false);
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  // Don't trust 3rd parties to ever load
```

Comment:
Clerk will fail to load for some users at times and we don't want completely brick the page with a loading state for them